model/authorization/TestTakerAuthorizationService.php

```diff
@@ -104,6 +113,34 @@ public function isProctored($deliveryId, User $user)
             : $deliverySyncService->isProctoredByDefault();
     }
 
+    /**
+     * @param string $deliveryId
+     * @param string $state
+     * @return bool
+     * @throws common_Exception
+     */
+    public function isActiveUnSecureDelivery($deliveryId, $state)
+    {
+        return $this->isNotSecure($deliveryId) && $state === DeliveryExecutionInterface::STATE_ACTIVE;
```

Comment:
It would be better in terms of performance to check constant value at first to do not call DB if $state is not equal `DeliveryExecutionInterface::STATE_ACTIVE`

Reply:
fixed